### ISSUE TITLE
Trailing comma rule flags multi-line method parameters

### ISSUE THREAD

**Issue by caesay** [palantir/tslint#2775]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.3.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
export default connect<SidebarStateProps, SidebarDispatchProps, SidebarOwnProps, SidebarProps>(
    mapStateToProps,
    mapDispatchToProps,
    mergeProps // <-- [tslint] Missing trailing comma (trailing-comma)
)(Sidebar);
```

with `tslint.json` configuration:

```json
"trailing-comma": [
    true, 
    {
        "multiline": "always", 
        "singleline": "never"
    }
],
```

#### Expected behavior
Shouldn't be an error if it's a method parameter list as this is invalid javascipt / typescript.

**Comment by adidahiya** [comment]
It's valid TypeScript, but invalid JavaScript. Duplicate of #2367

**Comment by ajafff** [comment]
@caesay Recently the rule got more configuration options. You can now ban trailing commas in parameter and/or arguments lists. https://palantir.github.io/tslint/rules/trailing-comma/

As @adidahiya noted, trailing commas are valid typescript.
Until JavaScript adds trailing commas you can have a different configuration for your JavaScript files:
```json
{
    "jsRules": {
        "trailing-comma": [true, {"multiline": {"arguments": "never", ...}}]
    }
}
```
